tube_rack/libraries/tuberack/source.rb

```diff
@@ -1,64 +1,105 @@
-# typed: false
 # frozen_string_literal: true
 
-module TubeRackHelper
-  def fetch_tube_rack(tube_rack)
-    show do
-      title 'Get Tube Rack'
-      note 'To help stay organized please get a'\
-           " <b>#{tube_rack.rack_type}</b>"
-      note "Make sure that the rack has #{tube_rack.rows} rows and #{tube_rack.columns} columns"
-      separator
-      note 'Using a small piece of tape label the <b>'\
-           "#{tube_rack.rack_type}</b> with <b>#{tube_rack.id}</b>"
-    end
-  end
 
-  def add_items(item_list, tube_rack)
-    item_list.each { |item| tube_rack.add_item(item) }
+# Factory class for instantiating `PlateLayoutGenerator`
+#
+# @author Devin Strickland <strcklnd@uw.edu>
+class TubeRackFactory
 
-    if tube_rack.is_a? TubeRack
-      tube_rack_instructions(item_list, tube_rack)
-    elsif tube_rack.is_a? StripWellRack
-      show do
-        note 'hey it worked thats wild'
-      end
+  def self.build(item_list:, object_type:, instructions: true)
+    rack = nil
+    if item_list.all? { |coll| coll.collection? }
+      rack = StripWellRack.new(object_type: object_type)
+    elsif item_list.all? { |item| item.is_a? Item }
+      rack = TubeRack.new(object_type: object_type)
     else
-      raise 'Unknown Tube Rack Class'
+      raise IncompatibleObjectType, 'Incompatible Object_Type'
     end
+    item_list.each { |item| rack.add_item(item) }
+    rack
   end
 
-  def tube_rack_instructions(item_list, tube_rack)
-    tube_rack.collection.get_non_empty.each_slice(3).to_a.each do |rc_chunk|
-      show do
-        title 'Place Samples into Tube Rack'
-        note "Place samples tubes in the <b>#{tube_rack.rack_type}</b>"\
-             ' per the table below'
-        table highlight_collection_rc(tube_rack.collection, rc_chunk) { |r, c|
-          tube_rack.part(r, c).id
-        }
-      end
-    end
-  end
 end
 
-
-
 class TubeRack
-
   TRUE_ITEM = 'true_item'.to_sym
 
-  attr_reader :rack_type, :collection, :rows, :columns, :id
+  attr_reader :object_type, :collection, :rows, :columns, :id
 
-  def initialize(rack_type:, collection: nil)
-    @collection = collection.nil? ? Collection.new_collection(rack_type) : collection
-    @rack_type = collection.nil? ? rack_type.to_s : collection.object_type.name
+  def initialize(object_type:, collection: nil)
+    @collection = collection.nil? ? Collection.new_collection(object_type) : collection
+    @object_type = collection.nil? ? object_type.to_s : collection.object_type.name
     @collection.mark_as_deleted
     @rows = @collection.dimensions[0]
     @columns = @collection.dimensions[1]
     @id = @collection.id
```

Comment:
I'm confused about the role of the collection here. There are many collections, then there is one, then it gets deleted, then you are using all its attributes. These classes should have a structure that parallels `MicrotiterPlate`, which is a wrapper for a collection that persists.